src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -33,12 +34,12 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
     @PutMapping("/{id}")
-    public BookDto update(@PathVariable Long id, @RequestBody BookDto bookDto) {
+    public BookDto update(@PathVariable Long id, @RequestBody @Valid BookDto bookDto) {
```

Comment:
You should provide CreateBookRequestDto or UpdateBookRequestDto instead of BookDto response